openrtx/src/rtx/OpMode_M17.cpp

```diff
@@ -335,6 +347,93 @@ void OpMode_M17::txState(rtxStatus_t *const status)
         modulator.sendFrame(m17Frame);
     }
 
+    if(lsfFragCount == 6)
+    {
+	   lsfFragCount = 0;
+       gpsStarted = false;
+
+       // reset metadata if no text message
+       if(last_text_blk == 0xff)
+       {
+           uint8_t buf[14];
+	       memset(buf, 0, 14);
+     	   lsf.setMetaText(buf);
+    	   type.fields.encSubType = M17_META_TEXT;
+    	   lsf.setType(type);
+           lsf.updateCrc();
+    	   encoder.encodeLsf(lsf, m17Frame);
+       }
+
+       // Wait at least 5 seconds between GPS transmissions
+       if((gpsTimer == -1 || gpsTimer >= 150))
+       {
+           gpsStarted = true;
+       }
+    }
+    else
+    	lsfFragCount++;
+
+
+    if(gpsStarted)
+    {
+        gpsStarted = false;
+        gpsTimer = 0;
+
+        gps_t gps_data;
+        pthread_mutex_lock(&state_mutex);
```

Comment:
Locking the state mutex in the rtx threads may introduce delays disrupting the correct timings for M17 modulation. GPS data is being fetched for read access and is actually safe to retrieve it without a critical section